yt_dlp/YoutubeDL.py

```diff
@@ -4074,6 +4105,12 @@ def urlopen(self, req):
                         'This request requires WebSocket support. '
                         'Ensure one of the following dependencies are installed: websockets',
                         cause=ue) from ue
+
+                elif re.match(r'unsupported (?:extensions: impersonate|impersonate target)', ue.msg.lower()):
+                    raise RequestError(
+                        f'Impersonate target "{compile_impersonate_target(*req.extensions["impersonate"])}" is not available.'
+                        f' This request requires browser impersonation, however you may be missing dependencies'
+                        f' required to support this target. See the documentation for more information.')
```

Comment:
I'm not sure if we want to hardcode the deps for impersonation in the code. There may be many factors at play why a target that was requested is not available:

- dep not installed
- dep installed but not a version that supports the target
- one impersonation dep installed but it doesn't support the target
 etc.

thinking of creating a spot on the README to describe the impersonation with examples, and list the deficiencies there.

Reply:
Are we planning to add more impersonation handlers asap? If not, I say we hardcode curl_cffi in message for now


> thinking of creating a spot on the README to describe the impersonation with examples, and list the deficiencies there.

Probably better in wiki.